### ISSUE TITLE
Broadcast notifications: JSON encode payload. Error code: 5

### ISSUE THREAD

**Issue by pultho** [laravel/framework#35322]
- Laravel Version: v7.29.3
- PHP Version: 7.4.5
- Database Driver & Version: MariaDB

### Description:

Broadcasting notifications seems to have trouble with invalid JSON content although returning the same content as JSON response doesn't throw issues. Does the framework miss any JSON or encoding options here?

### Steps To Reproduce:

Use notification broadcast channel with invalid JSON payload (we couldn't find any invalid characters in our data attributes yet).

### Stack trace:

`Illuminate\Queue\InvalidPayloadException: Unable to JSON encode payload. Error code: 5
#46 /vendor/laravel/framework/src/Illuminate/Queue/Queue.php(97): Illuminate\Queue\Queue::createPayload
#45 /vendor/laravel/framework/src/Illuminate/Queue/RedisQueue.php(91): Illuminate\Queue\RedisQueue::push
#44 /vendor/laravel/framework/src/Illuminate/Queue/Queue.php(46): Illuminate\Queue\Queue::pushOn
#43 /vendor/laravel/framework/src/Illuminate/Broadcasting/BroadcastManager.php(128): Illuminate\Broadcasting\BroadcastManager::queue`

### Queue:97

```
$payload = json_encode($this->createPayloadArray($job, $queue, $data));
     if (JSON_ERROR_NONE !== json_last_error()) 
         throw new InvalidPayloadException(
              'Unable to JSON encode payload. Error code: '.json_last_error()
        );
}

```
### Possible solutions:

Adding json encode options? Or encode the content before? How does the Laravel JSON response encode content?

**Comment by taylorotwell** [comment]
Have you tried different flags on your end to see if any particular thing fixes it?

**Comment by pultho** [comment]
Haven't had the chance yet since I would have to push that thing to production and wasn't able to recreate an environment for it locally.

But about the flags: _What flags does Laravel use for its default JSON responses?_ Since they never fail at all. 

I was checking for Symfony's JsonResponse to get some idea, it goes for: 

```
// Encode <, >, ', &, and " characters in the JSON, making it also safe to be embedded into HTML.
// 15 === JSON_HEX_TAG | JSON_HEX_APOS | JSON_HEX_AMP | JSON_HEX_QUOT
const DEFAULT_ENCODING_OPTIONS = 15;

protected $encodingOptions = self::DEFAULT_ENCODING_OPTIONS;
```



In general: JSON encoding options for HTTP responses and broadcasts should be the same - what do you think?

**Comment by pultho** [comment]
Yes, I went for different flags and pushed that thing to production (since I don't have a proper environment to re-create that case yet). 

In our Response.php we use the following encoding: `JSON_INVALID_UTF8_IGNORE | JSON_UNESCAPED_UNICODE`

I added this to Queue.php 97 (see above) which json_encodes the broadcast payload. `$payload = json_encode($this->createPayloadArray($job, $queue, $data), JSON_INVALID_UTF8_IGNORE | JSON_UNESCAPED_UNICODE);`

And the error was gone. But...

Queue is now going crazy unserializing the job again, throwing: `unserialize(): Error at offset 2109 of 44945 bytes`. So the payload still seems corrupt then.
